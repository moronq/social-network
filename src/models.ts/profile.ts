export type ProfileTypeResponse = {
  aboutMe: string
  userId: number | null
  lookingForAJob: boolean
  lookingForAJobDescription: string
  fullName: string
  contacts: ContactsType
  photos: {
    small: string
    large: string
  }
}

export type ProfileType = ProfileTypeResponse & { error: string }

type ContactsType = {
  github: string | null
  vk: string | null
  facebook: string | null
  instagram: string | null
  twitter: string | null
  website: string | null
  youtube: string | null
  mainLkink: string | null
}
