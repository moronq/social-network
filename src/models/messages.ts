export type ChateMessageType = {
  message: string
  photo: string
  userId: number
  userName: string
}
