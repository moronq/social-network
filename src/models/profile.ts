export type ProfileTypeResponse = {
  aboutMe: string
  userId: number | null
  lookingForAJob: boolean
  lookingForAJobDescription: string
  fullName: string
  contacts: ContactsType
  photos: PhotosType
}

export type PhotosType = {
  small: string
  large: string
}

export type ProfileTypeStatus = {
  status: string
}

export type ProfileType = ProfileTypeResponse & { error: string }

type ContactsType = {
  github: string | null
  vk: string | null
  facebook: string | null
  instagram: string | null
  twitter: string | null
  website: string | null
  youtube: string | null
  mainLkink: string | null
}

export type ProfileResponse<T = any> = {
  resultCode: number
  data: T
  fieldsError: Array<string>
  messages: Array<string>
}
