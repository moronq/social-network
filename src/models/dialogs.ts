import { PhotosType } from './profile'

export type dialogType = {
  id: number
  userName: string
  hasNewMessages: boolean
  lastDialogActivityDate: string
  lastUserActivityDate: string
  newMessagesCount: number
  photos: PhotosType
}
