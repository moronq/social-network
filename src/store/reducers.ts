import auth from './slices/authSlice/authSlice'
import profile from './slices/profileSlice/profileSlice'

export default {
  auth,
  profile,
}
