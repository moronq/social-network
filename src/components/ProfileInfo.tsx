import React, { FC, useState } from 'react'
import { ProfileType, ProfileTypeStatus } from '../models/profile'
import Button from './common/Button/Button'
import ProfileStatus from './ProfileStatus'

type PropsType = {
  profileInfo: ProfileType & ProfileTypeStatus
  userId: string
}

const ProfileInfo: FC<PropsType> = ({ profileInfo, userId }) => {
  const [details, setDetails] = useState(false)

  const showDetails = () => {
    setDetails(true)
  }
  const hideDetails = () => {
    setDetails(false)
  }
  return (
    <div className="border border-slate-300 grow flex flex-col rounded-lg p-2 bg-slate-100">
      <p className="text-xl">{profileInfo.fullName}</p>
      <ProfileStatus userId={userId} status={profileInfo.status} />
      {!details ? (
        <Button onClick={showDetails}>Show details</Button>
      ) : (
        <>
          <ul>
            <li>{profileInfo.aboutMe}</li>
            {profileInfo.lookingForAJob && <li>Looking for a job</li>}
            <li>{profileInfo.lookingForAJobDescription}</li>
          </ul>
          <Button onClick={hideDetails}>Hide</Button>
        </>
      )}
    </div>
  )
}

export default ProfileInfo
