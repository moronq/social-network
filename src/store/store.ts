import { combineReducers, configureStore } from '@reduxjs/toolkit'
import reducers from './reducers'

const rootReducer = combineReducers(reducers)

export const store = configureStore({
  reducer: rootReducer,
})
