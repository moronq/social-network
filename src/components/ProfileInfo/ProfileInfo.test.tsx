import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Provider } from 'react-redux'
import { ProfileType, ProfileTypeStatus } from '../../models/profile'
import { store } from '../../store/store'
import ProfileInfo from './ProfileInfo'

interface IMock {
  profileInfo: ProfileType & ProfileTypeStatus
  userId: string
}

const mockInfo: IMock = {
  profileInfo: {
    aboutMe: 'something about me',
    contacts: {
      github: 'github',
      vk: 'vk',
      facebook: 'facebook',
      instagram: 'instagram',
      twitter: 'twitter',
      website: 'website',
      youtube: 'youtube',
      mainLkink: 'mainLkink',
    },
    error: '',
    fullName: 'Sereja Shcherbinin',
    lookingForAJob: true,
    lookingForAJobDescription: 'I am searching for a job',
    photos: {
      large: '',
      small: '',
    },
    status: 'my status',
    userId: 1,
  },
  userId: '1',
}

describe('ProfileInfo test group', () => {
  test('ProfileInfo renders without errors', () => {
    render(
      <Provider store={store}>
        <ProfileInfo {...mockInfo} />
      </Provider>
    )
    expect(screen.getByText('Sereja Shcherbinin')).toBeInTheDocument
    expect(screen.getByText('my status')).toBeInTheDocument
    expect(screen.getByText('Show details')).toBeInTheDocument
    expect(screen.queryByText('something about me')).not.toBeInTheDocument
    expect(screen.queryByText('Hide')).not.toBeInTheDocument
  })
  test('ProfileInfo slider works', () => {
    render(
      <Provider store={store}>
        <ProfileInfo {...mockInfo} />
      </Provider>
    )
    expect(screen.queryByText('something about me')).not.toBeInTheDocument
    expect(screen.queryByText('Hide')).not.toBeInTheDocument
    expect(screen.getByText('Show details')).toBeInTheDocument

    userEvent.click(screen.getByRole('button'))
    expect(screen.queryByText('something about me')).toBeInTheDocument
    expect(screen.queryByText('Hide')).toBeInTheDocument
    expect(screen.queryByText('Show details')).not.toBeInTheDocument
  })
  test('ProfileInfo snapshot', () => {
    const element = render(
      <Provider store={store}>
        <ProfileInfo {...mockInfo} />
      </Provider>
    )
    expect(element).toMatchSnapshot()
  })
})
