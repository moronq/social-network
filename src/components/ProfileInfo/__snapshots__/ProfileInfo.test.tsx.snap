// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo test group ProfileInfo snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border border-slate-300 grow flex flex-col rounded-lg p-2 bg-slate-100"
      >
        <p
          class="text-xl"
        >
          Sereja Shcherbinin
        </p>
        <div>
          my status
        </div>
        <button
          class=" border border-slate-300 px-2 py-1 rounded-md bg-slate-100 hover:bg-slate-200"
        >
          Show details
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-slate-300 grow flex flex-col rounded-lg p-2 bg-slate-100"
    >
      <p
        class="text-xl"
      >
        Sereja Shcherbinin
      </p>
      <div>
        my status
      </div>
      <button
        class=" border border-slate-300 px-2 py-1 rounded-md bg-slate-100 hover:bg-slate-200"
      >
        Show details
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
